sktime/transformers/series_as_features/dictionary_based/_sax.py

```diff
@@ -166,29 +173,24 @@ def _generate_breakpoints(self):
         }[self.alphabet_size]
 
 
-class _BitWord:
+class _BitWord(object):
     # Used to represent a word for dictionary based classifiers such as BOSS
     # an BOP.
     # Can currently only handle an alphabet size of <= 4 and word length of
     # <= 16.
     # Current literature shows little reason to go beyond this, but the
     # class will need changes/expansions
     # if this is needed.
+    # TODO a shift of 2 is only correct for alphabet size 4, log2(4)=2
 
-    def __init__(self,
-                 word=0):
-        self.word = word
-
-    def push(self, letter):
-        # add letter to a word
-        self.word = (self.word << 2) | letter
+    @staticmethod
+    def create_bigram_word(word, other_word, length):
+        return (word << length) | other_word
 
-    def shorten(self, amount):
+    @staticmethod
+    def shorten_word(word, amount):
         # shorten a word by set amount of letters
-        self.word = self.right_shift(self.word, amount * 2)
-
-    def create_bigram(self, other_word, length):
-        return (self.word << length) | other_word.word
+        return _BitWord.right_shift(word, amount * 2)
```

Comment:
```suggestion
        return self.right_shift(word, amount * 2)
```

Reply:
Thank you for the suggestion. 

There was no self in the static methods to call self.right_shift. Thus, I have changed the methods to @classmethod such that I can call cls.right_shift(...)

```python    
   @classmethod
    def shorten_word(cls, word, amount):
        # shorten a word by set amount of letters
        return cls.right_shift(word, amount * 2)
```

Reply:
is that ok with you?

Reply:
Still seems complicated to me, I'd either just have standalone functions or a simple class, but maybe I'm missing something, happy to leave it as it is.  :)

Reply:
thank you :)